src/vm/program.ts

```diff
@@ -34,27 +35,99 @@ const Identifier = z.object({
   destination: z.string().optional(),
 });
 
-export type Identifier = z.infer<typeof Identifier>;
+const Identifiers = z.record(z.string(), Identifier);
+
+const FlowTrackingData = z.object({
+  ap_tracking: ApTrackingData,
+  reference_ids: z.record(z.string(), z.number()),
+});
+
+const Hint = z.object({
+  accessible_scopes: z.array(z.string()),
+  code: z.string(),
+  flow_tracking_data: FlowTrackingData,
+});
+
+const Hints = z.record(z.string(), z.array(Hint));
 
 const Program = z.object({
   attributes: z.any(),
   builtins: z.array(z.string()),
   compiler_version: z.string(),
   data: z
     .array(z.string())
-    .transform((value) => value.map((v) => new Felt(BigInt(v)))),
+    .transform((values) => values.map((v) => new Felt(BigInt(v)))),
   debug_info: z.any(), // TODO: DebugInfo
-  hints: z.record(z.string(), z.any()), // TODO: HintParams
-  identifiers: z
-    .record(z.string(), Identifier)
-    .transform((value) => new Map<string, Identifier>(Object.entries(value))),
+  hints: Hints,
+  identifiers: Identifiers,
   main_scope: z.string(),
   prime: z.string(),
   reference_manager: ReferenceManager,
 });
 
+export type ReferenceManager = z.infer<typeof ReferenceManager>;
+export type Reference = z.infer<typeof Reference>;
+export type ApTrackingData = z.infer<typeof ApTrackingData>;
+export type Identifier = z.infer<typeof Identifier>;
+export type Identifiers = z.infer<typeof Identifiers>;
+export type FlowTrackingData = z.infer<typeof FlowTrackingData>;
+export type Hint = z.infer<typeof Hint>;
+export type Hints = z.infer<typeof Hints>;
 export type Program = z.infer<typeof Program>;
+export class ProgramConstants extends Map<string, Felt> {
+  constructor(values: Array<[string, Felt]> = []) {
+    super(values);
+  }
+}
 
+/** Parse the program compilation artifacts to a Program object */
 export function parseProgram(program: string): Program {
   return Program.parse(JSON.parse(program));
 }
+
+/**
+ *  Parse the program identifiers
+ * to return a map of constant name to their value
+ */
+export function extractConstants(program: Program): ProgramConstants {
+  const constants: ProgramConstants = new ProgramConstants();
+  Object.entries(program.identifiers).map(([name, identifier]) => {
```

Comment:
don't do swith here, it's the same as the one in `findConstFromAlias`, so just map the resolving function, and `constants.set(...)`